src/renderers/dom/fiber/wrappers/ReactDOMFiberInput.js

```diff
@@ -209,8 +209,7 @@ var ReactDOMInput = {
           // browsers typically do this as necessary, jsdom doesn't.
           node.value = '' + value;
         }
-        // eslint-disable-next-line
-      } else if (value != node.value) {
+      } else if (node.value !== value) {
```

Comment:
shouldn't this be `node.value !== '' + value` to prevent unnecessary updates when you provide the same non-string value as before?

```js
render(<input value={0}/>, el);

render(<input value={0}/>, el); // shouldn't reset value
```